eradicate.py

```diff
@@ -28,176 +28,199 @@
 import io
 import os
 import re
-import sys
 import tokenize
 
 __version__ = '1.0'
 
+class Eradicator:
+    """Eradicate comments."""
+    MULTILINE_ASSIGNMENT_REGEX = re.compile(r'^\s*\w+\s*=.*[(\[{]$')
+    PARTIAL_DICTIONARY_REGEX = re.compile(r'^\s*[\'"]\w+[\'"]\s*:.+[,{]\s*$')
+
+    DEFAULT_WHITELIST = [
+        r'pylint',
+        r'pyright',
+        r'noqa',
+        r'type:\s*ignore',
+        r'fmt:\s*(on|off)',
+        r'TODO',
+        r'FIXME',
+        r'XXX'
+    ]
+    WHITELIST_REGEX = re.compile(r'|'.join(DEFAULT_WHITELIST), flags=re.IGNORECASE)
+
+    def comment_contains_code(self, line, aggressive=True):
+        """Return True comment contains code."""
+        line = line.lstrip()
+        if not line.startswith('#'):
+            return False
+
+        line = line.lstrip(' \t\v\n#').strip()
+
+        # Ignore non-comment related hashes. For example, "# Issue #999".
+        if re.search('#[0-9]', line):
+            return False
+
+        # Ignore whitelisted comments
+        if self.WHITELIST_REGEX.search(line):
+            return False
+
+        if re.match(r'.*?coding[:=][ \t]*([-_.a-zA-Z0-9]+)', line):
+            return False
+
+        # Check that this is possibly code.
+        for symbol in list('()[]{}:=%') + ['print', 'return', 'break', 'continue',
```

Comment:
This should be a constant too.

Reply:
I made it into constant `CODE_INDICATORS`